experimental/material_3_demo/lib/main.dart

```diff
@@ -52,18 +55,34 @@ class _AppState extends State<App> {
 
   void handleColorSelect(int value) {
     setState(() {
+      colorSelectionMethod = ColorSelectionMethod.colorSeed;
       colorSelected = ColorSeed.values[value];
     });
   }
 
+  void handleImageSelect(int value) {
+    setState(() {
+      colorSelectionMethod = ColorSelectionMethod.image;
+      imageSelected = ColorImageProvider.values[value];
+      ColorScheme.fromImageProvider(
+              provider: NetworkImage(ColorImageProvider.values[value].url))
+          .then((newScheme) => imageColorScheme = newScheme);
```

Comment:
Should we defer changing the app's state until the image loads and the new color scheme has been computed?
```dart
void handleImageSelect(int value) {
  final String url = ColorImageProvider.values[value].url;
  ColorScheme.fromImageProvider(
    provider: NetworkImage(url).then((newScheme) {
      setState(() {
        colorSelectionMethod = ColorSelectionMethod.image;
        imageSelected = ColorImageProvider.values[value];
        imageColorScheme = newScheme;
      }),
    }),
  );
}
```

Reply:
Good shout, thanks